src/main/scheduler/scheduler.c

```diff
@@ -224,25 +205,32 @@ void scheduler(void)
     const timeUs_t currentTimeUs = micros();
 
     // Check for realtime tasks
-    timeUs_t timeToNextRealtimeTask = TIMEUS_MAX;
-    for (const cfTask_t *task = queueFirst(); task != NULL && task->staticPriority >= TASK_PRIORITY_REALTIME; task = queueNext()) {
-        const timeUs_t nextExecuteAt = task->lastExecutedAt + task->desiredPeriod;
-        if ((int32_t)(currentTimeUs - nextExecuteAt) >= 0) {
-            timeToNextRealtimeTask = 0;
-        } else {
-            const timeUs_t newTimeInterval = nextExecuteAt - currentTimeUs;
-            timeToNextRealtimeTask = MIN(timeToNextRealtimeTask, newTimeInterval);
-        }
-    }
-    const bool outsideRealtimeGuardInterval = (timeToNextRealtimeTask > 0);
+	bool outsideRealtimeGuardInterval = 1;
+	
+	//task
+	cfTask_t *task;
+	
+	for (int taskQueuePos = 0; taskQueuePos < taskQueueSize; taskQueuePos++)
+	{
+		task = taskQueueArray[taskQueuePos];
+		if (task->staticPriority < TASK_PRIORITY_REALTIME)
+		{
+			break;
+		}
+		if (currentTimeUs > task->lastExecutedAt + task->desiredPeriod) {
+			outsideRealtimeGuardInterval = 0;
+			break;
+		}
+	}
```

Comment:
Your new calculation of the `outsideRealtimeGuardInterval` is indeed a speed improvement. However the test `currentTimeUs > task->lastExecutedAt + task->desiredPeriod` doesn't take into account that, on some targets `currentTimeUs` is of type `uint64_t` (this is true for most of the iNav targets which share this code). So `if ((timeDelta_t)(currentTimeUs - task->lastExecutedAt) >= task->desiredPeriod)` is a better comparison.

Also changing the queue iterator to directly use an array index decreases the readability of the code without bringing any performance improvement, since the compiler inlines the `queueFirst` and `queueNext` calls. So I suggest the first loop is replaced by:
```
    // Check for realtime tasks
    bool outsideRealtimeGuardInterval = true;
    for (const cfTask_t *task = queueFirst(); task != NULL && task->staticPriority >= TASK_PRIORITY_REALTIME; task = queueNext()) {
        if ((timeDelta_t)(currentTimeUs - task->lastExecutedAt) >= task->desiredPeriod) {
            outsideRealtimeGuardInterval = false;
            break;
        }
    }
```

Note for future changes: please ensure K&R convention for braces is used, and use `true` and `false` instead of `1` and `0` for boolean values.


However changing the queue iterator to use a direct index into the array decrease